src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,302 @@
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metaObj =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = file[0] + hashFile[0].split('.')[0];
+        for (const path in metaObj) {
+          if (flattenMap[path] === undefined) {
+            const uri = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path);
+            await Metadata.disable(uri, hash);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * A Singleton Object
+ *
+ * PathToHash is a data structure that stores the path of the actual data
+ * and the hash value generated by the contents of the actual data
+ * to synchronize the actual data and metadata
+ */
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+  public static startFlag: boolean = false;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      PathToHash.startFlag = true;
+    }
+    return this._instance;
+  }
+
+  private async init() {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const uri = vscode.workspace.workspaceFolders[0].uri;
+    const map = await this.scanRecursively(uri);
+    await MetadataSynchronizer.run(await this.getFlatMap(map));
+    return map;
+  }
+
+  private async scanRecursively(uri: vscode.Uri) {
+    let subMap: {[key: string]: any} = {};
+    const files = await vscode.workspace.fs.readDirectory(uri);
+
+    for (const file of files) {
+      const name: string = file[0];
+      const type: number = file[1];
+
+      if (type === vscode.FileType.File) {
+        if (vscode.workspace.workspaceFolders !== undefined) {
+          if (isOneExplorerTargetFile(
+                  vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, name))) {
+            subMap[name] = await generateHash(vscode.Uri.joinPath(uri, '/' + name));
+          }
+        }
+      } else if (type === vscode.FileType.Directory && name !== '.one-vscode') {
+        const temp = await this.scanRecursively(vscode.Uri.joinPath(uri, '/' + name));
+        if (temp !== undefined) {
+          subMap[name] = temp;
+        }
+      }
+    }
+    if (Object.keys(subMap).length === 0) {
+      return {};
+    }
+
+    return subMap;
+  }
+
+  private async getFlatMap(map: any) {
+    let flatMap: any = {};
+    let queue = [];
+    for (let data in map) {
+      queue.push([map[data], data.toString()]);
+    }
+    while (queue.length !== 0) {
+      const node: any = queue.pop();
+      if (node === undefined) {
+        continue;
+      }
+      if (vscode.workspace.workspaceFolders === undefined) {
+        break;
+      }
+      let path = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, node[1]).path;
```

Comment:
```suggestion
      const path = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, node[1]).fsPath;
```
- As path not changes, let's use const
- AFAIK,, 'fsPath' is for filesystem path and 'path'is the `/some/path` part of `http://www.example.com/some/path?query#fragment`